src/ucp/wireup/wireup.c

```diff
@@ -763,6 +755,12 @@ static ucs_status_t ucp_wireup_msg_handler(void *arg, void *data,
                 if (msg->type != UCP_WIREUP_MSG_EP_CHECK) { goto out; },
                 "WIREUP message (%d src_ep_id 0x%" PRIx64 " sn %d)", msg->type,
                 msg->src_ep_id, msg->conn_sn);
+
+        if ((msg->type == UCP_WIREUP_MSG_EP_CHECK) && (ep != NULL)) {
+            /* UCP EP is valid, no need for any other actions when handling
+             * EP_CHECK wireup message */
+            goto out;
```

Comment:
IMO previous code was better: ucp_wireup_process_ep_check was making the decision of whether to send the EP_REMOVED message or not. which was inline with other message types.
After this change, the logic of this decision is moved to here. So ucp_wireup_process_ep_check is actually "
"ucp_wireup_send_ep_removed"

Reply:
yes, but do we need to spend time unpacking the address?
also, IMO the function was error prone, because we had two EPs (one of them is always `NULL`).
so, we tried to destroy the `ep` there which is `NULL`. it happens because almost all other functions has variable/parameter `ep`/`ucp_ep`

Reply:
ok, so just rename to ucp_wireup_send_ep_removed + comment that we avoid unpacking the address

Reply:
fixed

Reply:
@yosefe PR is small enough, force-pushed changes